### PR TITLE
Trade

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: include/decompress.h [hunk 1]
```diff
@@ -23,7 +23,6 @@ void DecompressPicFromTable_2(const struct CompressedSpriteSheet *src, void* buf
 void DecompressPicFromTable_DontHandleDeoxys(const struct CompressedSpriteSheet *src, void* buffer, s32 species);
 
 void HandleLoadSpecialPokePic(const struct CompressedSpriteSheet *src, void *dest, s32 species, u32 personality);
-void HandleLoadSpecialPokePic_2(const struct CompressedSpriteSheet *src, void *dest, s32 species, u32 personality);
 void HandleLoadSpecialPokePic_DontHandleDeoxys(const struct CompressedSpriteSheet *src, void *dest, s32 species, u32 personality);
 void HandleLoadSpecialPokePic(const struct CompressedSpriteSheet *src, void *dest, s32 species, u32 personality);
```
Comment (ultima-soul) [inline include/decompress.h]: Duplicate declaration for HandleLoadSpecialPokePic.

File: src/data/ingame_trades.h [hunk 2]
```diff
@@ -0,0 +1,144 @@
+const struct InGameTrade gInGameTrades[] = {
+    {
+        .nickname = _("MIMIEN"),
+        .species = SPECIES_MR_MIME,
+        .ivs = {20, 15, 17, 24, 23, 22},
+        .abilityNum = FALSE,
```
Comment (ultima-soul) [inline src/data/ingame_trades.h]: abilityNum isn't a boolean.

File: src/trade_scene.c [hunk 3]
```diff
@@ -0,0 +1,2859 @@
...
+        {
+            sTradeData->state++;
+            SetTradeSequenceBgGpuRegs(1);
+            sTradeData->bg1vofs = 166;
+            SetTradeSequenceBgGpuRegs(3);
+            sTradeData->bg2vofs = 412;
+            sTradeData->unk_90 = CreateSprite(&gUnknown_826CE44, 120, -20, 3);
+            gSprites[sTradeData->unk_90].callback = SpriteCB_TradeGlowWireless;
+            sTradeData->unk_91 = CreateSprite(&sGlowBallSpriteTemplate, 120, -20, 0);
+            StartSpriteAnim(&gSprites[sTradeDa
```
Comment (jiangzhengwenjz) [inline src/trade_scene.c]: Inconsistent pointer space style

File: src/trade.c [hunk 4]
```diff
@@ -0,0 +1,4501 @@
...
+static void sub_804F020(u8 side);
+static void sub_804F08C(u8 a0, u8 partyIdx, u8 a2, u8 a3, u8 a4, u8 a5);
+static void sub_804F284(u8 side);
+static void sub_804F3B4(void);
+static void sub_804F3C8(u8 a0);
+static void TradeMenuAction_Summary(u8 taskId);
+static void TradeMenuAction_Trade(u8 taskId);
+static void ScheduleLinkTaskWithDelay(u16 delay, u8 kind);
+static void RunScheduledLinkTasks(void);
+static void PrintTradeErrorOrStatusMessage(u8 str_idx);
```
Comment (jiangzhengwenjz) [inline src/trade.c]: strIdx

File: src/trade.c [hunk 5]
```diff
@@ -0,0 +1,4501 @@
...
+    0x12, 0x05, 0x12, 0x07,
+    0x08, 0x07, 0x16, 0x0c,
+    0x08, 0x07, 0x16, 0x0c,
+    0x06, 0x07, 0x18, 0x0c,
+    0x06, 0x07, 0x18, 0x0c,
+    0x08, 0x07, 0x16, 0x0c,
+    0x07, 0x07, 0x17, 0x0c
+};
+
+const u8 gUnknown_8261EB6[] = _("");
```
Comment (jiangzhengwenjz) [inline src/trade.c]: Are you sure these are not static?

File: src/trade.c [hunk 6]
```diff
@@ -0,0 +1,4501 @@
...
+    sizeof(struct SaveBlock1),
+    sizeof(struct MapData),
+    0x530, // unk
+    0x34, // unk
+    sizeof(struct MailStruct),
+    sizeof(struct Pokemon),
+    0x528 // unk
+};
+
+static const u16 gTradeMovesBoxTilemap[] = INCBIN_U16("graphics/trade/moves_box_map.bin");
```
Comment (jiangzhengwenjz) [inline src/trade.c]: static symbols should start with prefix `s` if documented (please also check other instances in this file)

File: src/trade.c [hunk 7]
```diff
@@ -0,0 +1,4501 @@
...
+    case 7:
+        CalculateEnemyPartyCount();
+        SetGpuReg(REG_OFFSET_DISPCNT, 0);
+        SetGpuReg(REG_OFFSET_BLDCNT, 0);
+        sTradeMenuResourcesPtr->partyCounts[0] = gPlayerPartyCount;
+        sTradeMenuResourcesPtr->partyCounts[1] = gEnemyPartyCount;
+
+        for (i = 0; i < sTradeMenuResourcesPtr->partyCounts[0]; i++)
+        {
+            struct Pokemon *mon = &gPlayerParty[i];
```
Comment (jiangzhengwenjz) [inline src/trade.c]: Inconsistent pointer space style

File: src/trade.c [hunk 8]
```diff
@@ -0,0 +1,4501 @@
...
+                                                                (gTradeMonSpriteCoords[i][0] * 8) + 14,
+                                                                (gTradeMonSpriteCoords[i][1] * 8) - 12,
+                                                                1,
+                                                                GetMonData(mon, MON_DATA_PERSONALITY),
+                                                                TRUE);
+        }
+
+        for (i = 0; i < sTradeMenuResourcesPtr->partyCounts[1]; i++)
+        {
+            struct Pokemon *mon = &gEnemyParty[i];
```
Comment (jiangzhengwenjz) [inline src/trade.c]: Inconsistent pointer space style

File: src/trade.c [hunk 9]
```diff
@@ -0,0 +1,4501 @@
...
+    case 0:
+        sub_804C600();
+        gMain.state++;
+        break;
+    case 1:
+        gMain.state++;
+        sTradeMenuResourcesPtr->unk_A8 = 0;
+        break;
+    case 2:
+        gMain.stat
```
Comment (jiangzhengwenjz) [inline src/trade.c]: Inconsistent pointer space style

Comment (jiangzhengwenjz) [inline src/trade.c]: Inconsistent pointer space style

Comment (jiangzhengwenjz) [inline src/trade.c]: Inconsistent pointer space style

Comment (jiangzhengwenjz) [inline src/trade.c]: For function call in multiple lines please be consistent with the location of first argument. e.g. line 857
And please check other instances in this file

Comment (jiangzhengwenjz) [inline src/trade.c]: JOY_NEW(A_BUTTON)

File: asm/battle_tower.s [hunk 10]
```diff
@@ -2832,7 +2832,7 @@ _080E6F68:
 	ldrb r1, [r0]
 	movs r0, 0x64
 	muls r0, r1
-	ldr r1, _080E6FB4 @ =gUnknown_2024220
+	ldr r1, _080E6FB4 @ =gEnemyParty + 0x1f4
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply asm/battle_tower.s]: Yes and yes. Also, what other instances? battle_tower is the only instance of `gEnemyParty + fixed_byte_offset` left.

[the remark this answers is not in the document]
Comment (jiangzhengwenjz) [reply asm/battle_tower.s]: At least there's `gPlayerParty + 0x64` in asm/party_menu.s

File: src/trade.c (hunk 9 above)
Comment (PikalaxALT) [reply src/trade.c]: Leaving this until the functions in question can be matched

File: asm/battle_tower.s (hunk 10 above)
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply asm/battle_tower.s]: Oh right, forgot about those lol

File: src/trade.c (hunk 5 above)
Comment (PikalaxALT) [reply src/trade.c]: Surrounding strings are not static.

File: src/trade_scene.c [hunk 11]
```diff
@@ -64,18 +74,18 @@ struct TradeAnimationResources {
...
+    /*0x8A*/ u16 linkTimeoutTimer;
     /*0x8C*/ u16 unk_8C;
     /*0x8E*/ u8 pokePicSpriteIdxs[2];
-    /*0x90*/ u8 unk_90;
-    /*0x91*/ u8 unk_91;
-    /*0x92*/ u8 unk_92;
-    /*0x93*/ u8 unk_93;
+    /*0x90*/ u8 tradeGlow1SpriteId;
+    /*0x91*/ u8 gbaScreenSpriteID;
+    /*0x92*/ u8 linkCableEndSpriteId;
```
Comment (jiangzhengwenjz) [inline src/trade_scene.c]: Is there a reason to use both ID and Id?

File: src/trade_scene.c [hunk 12]
```diff
@@ -116,58 +126,58 @@ static void SpriteCB_TradeGlowWireless(struct Sprite * sprite);
...
-static void sub_80543C4(void);
-static void sub_8054470(u8 taskId);
+static void LinkTrade_TearDownAssets(void);
+static void Task_WaitFadeAndStartInGameTradeAnim(u8 taskId);
 static void CheckPartnersMonForRibbons(void);
 static void Task_AnimateWirelessSignal(u8 taskId);
-static void c3_0805465C(u8 taskId);
-static void sub_8054734(u8 taskId);
+static void Task_OpenCetnerWhiteColumn(u8 taskId);
+static void Task_CloseCetnerWhiteColumn(u8 taskId);
```
Comment (jiangzhengwenjz) [inline src/trade_scene.c]: Is cetner a common english word?

File: src/trade_scene.c [hunk 13]
```diff
@@ -415,28 +425,28 @@ static const union AnimCmd *const gSpriteAnimTable_826CF24[] = {
...
+    .tileTag = TAG_GBA_SCREEN_TILES,
+    .paletteTag = TAG_GBA_PAL,
     .oam = &gOamData_826CED0,
     .anims = gSpriteAnimTable_826CF24,
     .affineAnims = gDummySpriteAffineAnimTable,
-    .callback = sub_804FE00
+    .callback = SpriteCB_TradeGBAScreen
 };
 
 static const u16 gTradeGlow2PaletteAnimTable[] = {
```
Comment (jiangzhengwenjz) [inline src/trade_scene.c]: static symbol should start with prefix `s`if documented

**Pull request merged**